### DISCUSSION TITLE
JSON log parser - Scope array support

### DISCUSSION THREAD

**Discussion by bednart** [grafana/grafana#47389] [Dashboards, Panels & Visualizations]
Hi,

I know that there is "Arrays are skipped" note at [log queries](https://grafana.com/docs/loki/latest/logql/log_queries/#json) documentation but I would like to ask why is that? And would it be possible to add this support?

We are migrating from Elastic and Kibana to Grafana Cloud and we are using log scopes in .NET C#:

![image](https://user-images.githubusercontent.com/10383704/161962610-a8c083be-6329-43d5-a149-8243dca5beeb.png)

In Kibana all scope fields are parsed on stored as separated fields so it is easily filtered:

![image](https://user-images.githubusercontent.com/10383704/161963123-99e91a69-670c-4f3d-97ef-86b6879e2612.png)

But in Grafana cloud scope is there but using | json log query it is not parsed as label on client. (Ignore different HealtCheckName :))

![image](https://user-images.githubusercontent.com/10383704/161963476-d9da5e17-64af-4957-b1a7-a3b0e9073c63.png)

It would be nice to parse scope array in same way as state object is. Somethink like this:

- Scopes_0_HealtCheckName
- Scopes_0_HealthStatus

What do you think?

Thanks,
beda

**Comment by gabor** [comment]
hi @bednart , do i understand this correctly, that you'd like to change something in the loki LogQL langauge's `json` filter's behavior?

if it is so, i recommend creating an issue in Loki's github repo, at https://github.com/grafana/loki

**Reply by bednart** [reply]
There is an issue here https://github.com/grafana/loki/issues/8475

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
